lib/config/configuration.go

```diff
@@ -1903,17 +1903,22 @@ func applyAppsConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 			}
 		}
 
+		if application.AlwaysUseProxyPublicAddr && application.PublicAddr != "" {
+			return trace.BadParameter("public_addr cannot be explicitly set if always_use_proxy_public_addr is set to true")
+		}
+
```

Comment:
this isn't necessarily required but I decided to do it this way since the public_addr is an optional field. If a customer explicitly sets the public addr, i don't think overriding it with this field makes a bunch of sense. I also wanted this field to be implied based on the non-existence of the `public_addr` instead of set but, its better to be explicit with both